02.calendar/cal.rb

```diff
@@ -0,0 +1,48 @@
+require "date"
+require "optparse"
+
+opt = OptionParser.new
+opt.on("-y [VAL]") {|year| @year = year}
+opt.on("-m [VAL]") {|month| @month = month}
+opt.parse(ARGV)
+
+if @year != nil
+  @year
+else
+  @year = Date.today.year
+end
```

Comment:
`@year` に値が入っていた時に何もしていないので、

```suggestion
unless @year
  @year = Date.today.year
end
```

こんな感じで「@year に値が入っていなかった時」の処理だけにするとスッキリしますよ。
下の `@month` も同様です。